server/src/controller/user.controller.js

```diff
@@ -130,8 +133,20 @@ exports.querylogin = (req, res) => {
             res.json({
```

Comment:
The reason your cookie isn't being set in the browser is because you're sending the response before adding the cookie header.

```
res
.cookie("token", token, {
  maxAge: 1000 * 60 * 10,
  httpOnly: true
})
.json({
  success: true,
  _id: user._id,
  token: token
});
```

You can restructure your code to first set a cookie header, then send the response. That should work.

Right now your token will still be `undefined`, which is your next problem to solve :)

Reply:
Also note there is no reason to send the token as a response to the user - it defeats the purpose of having an `httpOnly` cookie. Your token is in the cookie.

Reply:
thanks a lot Michiel for clarification, I will try and update the code.